retrace/glstate_shaders.cpp

```diff
@@ -304,6 +304,105 @@ struct VertexAttribDesc : AttribDesc
     }
 };
 
+/* code for converting fp11 and fp10 values to fp32 is taken from Mesa:
+ *
+ * The* Copyright (C) 2011 Marek Olšák <maraeo@gmail.com>
```

Comment:
It's the same license (MIT), so instead to duplicating the license, please just add `Copyright (C) 2011 Marek Olšák <maraeo@gmail.com>` to the top of this file, below Jose Fonseca.

Another alternative, is to put these helpers on their own header file.  Probably a good idea on its own right.

Reply:
I did the latter and added a header file in helpers with these functions.